### PR TITLE
Don't leak Mix.shell between tests

### PR DESCRIPTION
**Pull request by wojtekmach**
Ref https://github.com/elixir-lang/elixir/issues/7898#issuecomment-404922002

### CODE REVIEW

**Comment by wojtekmach** [comment]
CI still broken 😢 help appreciated!

**Comment by josevalim** [comment]
@wojtekmach it sucks we can no longer make those tests async. :( Maybe we need a solution for this in Elixir itself. Unsure.

**Comment by OvermindDL1** [comment]
No way to mess around with the group leaders to make distinct groupings for it while being async?  More processes needed then most likely.

File: test/support/mix_task_case.ex [hunk 1]
```diff
@@ -0,0 +1,10 @@
+defmodule Ecto.MixTaskCase do
+  use ExUnit.CaseTemplate
+
+  setup_all do
```
Comment (ericmj) [inline test/support/mix_task_case.ex]: Shouldn't this be `setup do` so that each test resets global state?

**Comment by ericmj** [comment]
> @wojtekmach it sucks we can no longer make those tests async. :( Maybe we need a solution for this in Elixir itself. Unsure.

We could make it Registry based to that each process can have its own state. Mix.Process sends to self anyway.

**Comment by josevalim** [comment]
@ericmj I was thinking maybe something simpler, like `ExUnit.after_suite` callback. Thoughts?

**Comment by josevalim** [comment]
@ericmj after the suite is over we will revert the Mix shell back to the IO one, before running cover.

**Comment by josevalim** [comment]
Closing in favor of https://github.com/elixir-lang/elixir/issues/8082

**Pull request closed without merging**